### PR TITLE
Included endogenous labor supply and S periods

### PR DESCRIPTION
**Pull request by adibalaram**
I've modified the households.py and SS.py files to incorporate both endogenous labor supply and S-periods. The earlier version of households.py didn't simultaneously account for all 2S-1 FOCs. I have also fixed some typos and cleaned up the code a bit.

### CODE REVIEW

**Comment by LeiliPR** [comment]
Thanks Aditia, endogenous labor supply was my concern from yesterday. I was confused with the first household.py work as it had exogenous labor.

File: scripts/SS.py [hunk 1]
```diff
@@ -18,10 +18,13 @@ def solve_ss(r_init, w_init, params):
     w = w_init
     while (ss_dist > ss_tol) & (ss_iter < ss_max_iter):
         # solve HH problem
-        foc_args = (beta, sigma, r, w, n, 0.0)
-        b_sp1_guess = [0.05, 0.05]
-        result = opt.root(hh.FOCs, b_sp1_guess, args=foc_args)
-        b_sp1 = result.x
+        foc_args = (beta, sigma, r, w, 0.0)
+        r_s_guess = np.ones(S)
```
Comment (jdebacker) [inline scripts/SS.py]: should this be `n_s_guess`?

Comment (adibalaram) [reply scripts/SS.py]: Yes. Thanks for pointing that out.

**Comment by jdebacker** [comment]
@adibalaram is this now ready to merge?

**Comment by adibalaram** [comment]
Yes, it is.

**Pull request merged**